jars/sources/ExerelinCore/exerelin/campaign/MiningHelperLegacy.java

```diff
@@ -954,9 +943,126 @@ public static MiningResult getMiningResults(CampaignFleetAPI fleet, SectorEntity
 			
 			applyResourceExhaustion(entity, strength);
 		}
-		
+
+		if(enableMiningHarass){
+			float miningValue = computeMiningValue(result);
+
+			float miningHarass = Global.getSector().getMemoryWithoutUpdate().getFloat(MINING_HARASS_KEY);
+
+			if(miningHarass + miningValue > miningHarassThreshold || DEBUG_MODE ){
+				Global.getSector().getMemoryWithoutUpdate().set(MINING_HARASS_KEY, 0);
+				spawnFleet(getHostileFaction());
+			}else{
+				Global.getSector().getMemoryWithoutUpdate().set(MINING_HARASS_KEY, miningHarass + miningValue);
+			}
+		}
+
 		return result;
 	}
+
+	public static float computeMiningValue(MiningResult result){
+		float miningValue = 0f;
+
+		for (Map.Entry<String, Float> entry : result.report.totalOutput.entrySet()) {
+			miningValue += Global.getSector().getEconomy().getCommoditySpec(entry.getKey()).getBasePrice() * entry.getValue();
+		}
+
+		for(CacheResult cacheResult : result.cachesFound){
+			switch(cacheResult.def.type){
+				case COMMODITY :
+					miningValue += Global.getSector().getEconomy().getCommoditySpec(cacheResult.def.id).getBasePrice() * cacheResult.numItems;
+					break;
+				case HULLMOD :
+					miningValue += Global.getSettings().getHullModSpec(cacheResult.def.id).getBaseValue() * cacheResult.numItems;
+					break;
+				case FRIGATE :
+					miningValue += Global.getSettings().getHullSpec(cacheResult.def.id).getBaseValue() * cacheResult.numItems;
+					break;
+				case FIGHTER_WING :
+					miningValue += Global.getSettings().getFighterWingSpec(cacheResult.def.id).getBaseValue() * cacheResult.numItems;
+					break;
+				case WEAPON :
+					miningValue += Global.getSettings().getWeaponSpec(cacheResult.def.id).getBaseValue() * cacheResult.numItems;
+					break;
+			}
+		}
+
+		return miningValue;
+	}
+
+	protected static void spawnFleet(FactionAPI faction){
```

Comment:
This function should be moved out of "MiningHelperLegacy.java" so it can be reused when another feature requires "spawning fleets".